core/arch/arm/kernel/thread_spmc.c

```diff
@@ -640,6 +640,20 @@ static int add_mem_share_frag(struct mem_frag_state *s, void *buf, size_t flen)
 	return rc;
 }
 
+static bool is_sp_share(void *buf)
+{
+	struct ffa_mem_transaction *input_descr = NULL;
+	struct ffa_mem_access_perm *perm = NULL;
+
+	if (!IS_ENABLED(CFG_SECURE_PARTITION))
+		return false;
+
+	input_descr = buf;
+
+	perm = &input_descr->mem_access_array[0].access_perm;
+	return READ_ONCE(perm->endpoint_id) != my_endpoint_id;
```

Comment:
`perm->endpoint_id` is read again in `add_mem_region_to_sp()` called from `spmc_sp_add_share()`.

Reply:
Here it's only used to check if the share is intended for the OP-TEE endpoint.

Reply:
Yes, the sanitation performed in `add_mem_region_to_sp()` should make it safe to read it twice. I might be worth adding a comment about this somewhere.

Reply:
Added a comment